storage/remote/codec.go

```diff
@@ -540,3 +542,14 @@ func labelsToLabelsProto(labels labels.Labels, buf []prompb.Label) []prompb.Labe
 	}
 	return result
 }
+
+// metricTypeToMetricTypeProto transforms a Prometheus metricType into prompb metricType. Since the former is a string we need to transform it to an enum.
+func metricTypeToMetricTypeProto(t textparse.MetricType) prompb.MetricMetadata_MetricType {
+	mt := strings.ToLower(string(t))
```

Comment:
I don't think this works. 

The fields are:

```
var MetricMetadata_MetricType_value = map[string]int32{
	"Unknown":        0,
	"Counter":        1,
	"Gauge":          2,
	"Histogram":      3,
	"GaugeHistogram": 4,
	"Summary":        5,
	"Info":           6,
	"StateSet":       7,
}
```

since we are lowercasing, this should never match anything. This also means we don't have enough unit tests as we did not detect that (: 


Once my comment for being consistent with style guide, thus making those fields upper case, we can do `ToUppper` here and that might work then. Still unit test would be nice.

Reply:
You're absolutely right, fantastic catch. It went out of sync when I changed originally to camel case. Initially, I didn't think this needed tests, I was wrong and have added them now.

Reply:
Not seeing where this was resolved either?